tests/s3/versioning/test_versioning_multipart.py

```diff
@@ -0,0 +1,327 @@
+        res = self.s3_ver_test_obj.put_bucket_versioning(bucket_name=self.bucket_name,
+                                                         status="Suspended")
+        assert_utils.assert_true(res[0], res[1])
+        if os.path.exists(self.file_path):
+            os.remove(self.file_path)
+        create_file(self.file_path, mp_config["file_size"])
+        precalc_parts = s3_utils.get_precalculated_parts(self.file_path, mp_config["part_sizes"],
+                                                         chunk_size=mp_config["chunk_size"])
+        self.log.info("Step 3-5: Upload multipart object to a bucket")
+        mpu_id, resp, sorted_lst = self.initiate_upload_list_mpu(self.bucket_name, self.object_name,
+                                                                 is_part_upload=True,
+                                                                 is_lst_mpu=True,
+                                                                 parts=precalc_parts)
+        self.log.info("Step 6: List multipart uploads")
+        res = self.s3_mp_test_obj.list_multipart_uploads(self.bucket_name)
+        assert_utils.assert_not_in("VersionId", res[1])
+        self.log.info("Step 7: Complete multipart upload")
+        _, resp = self.s3_mp_test_obj.complete_multipart_upload(mpu_id, sorted_lst,
+                                                                self.bucket_name, self.object_name)
+        #assert_utils.assert_in("VersionId", resp)
+        assert_utils.assert_in("ETag", resp)
+        self.log.info("Step 8: List Object Versions")
+        _, list_resp = check_list_object_versions(self.s3_ver_test_obj,
+                                                  bucket_name=self.bucket_name,
+                                                  expected_versions={}, )
+        versions = list_resp[1]["Versions"][0]
+        assert_utils.assert_equal(versions["VersionId"], "null", "VersionId is not None")
+        assert_utils.assert_equal(str(versions["IsLatest"]), "True", "IsLatest is not True")
+        assert_utils.assert_in("ETag", versions.keys(), "ETag is not available")
+        self.log.info("Step 9: Check GET/HEAD Object")
+        check_get_head_object_version(self.s3_test_obj, self.s3_ver_test_obj,
+                                      bucket_name=self.bucket_name,
+                                      object_name=self.object_name)
+        self.log.info("Step 10: Check GET/HEAD Object with versionId = null")
+        check_get_head_object_version(self.s3_test_obj, self.s3_ver_test_obj,
+                                      bucket_name=self.bucket_name,
+                                      object_name=self.object_name,
+                                      version_id="null")
+        self.log.info("ENDED: Test multipart upload in a versioning suspended bucket.")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-41287")
+    @CTFailOn(error_handler)
+    def test_preexist_mpu_versioning_enabled_bkt_41287(self):
```

Comment:
Please update the test method name and the test logic to use upload_version (with changes for multipart from @DhobaleVishal's PR), delete_version and existing common libs

Reply:
I am not sure if that is tested fully yet. Will check and refactor if its working for him

Reply:
Done